iiblib/iib_build_details_model.py

```diff
@@ -1,135 +1,336 @@
+
+    def __init__(self, *args, **kwargs):
+        self._data = self._get_args(kwargs)
+
+    @classmethod
+    def _validate_data(cls, request_types_and_models, data):
         """
+        Validate data with class accepted request type
+
         Args:
-            _id (int)
-                Id of build
-            state (str)
-                State of build
-            state (str)
-                Reason for state change
-            from_index (str)
-                Reference of index image used as source for rebuild
-            from_index_resolved (str)
-                Reference of new index image
-            bundles (list)
-                List of bundles to be added to index image
-            removed_operators (list)
-                List of operators to be removed from index image
-            organization (str)
-                Name of organization to push to in the legacy app registry
-            binary_image (str)
-                Reference of binary image used for rebuilding
-            binary_image_resolved (str)
-                Checksum reference of binary image that was used for rebuilding
-            index_image (str)
-                Reference of index image to rebuild
-            request_type (str)
-                Type of iib build task (add or remove)
-            arches (list)
-                List of architectures supported in new index image
-            bundle_mapping (dict)
-                Operator names in "bundles" map to: list of "bundles" which
-                map to the operator key
-            omps_operator_version (dict)
-                Operator version returned from OMPS API call used for Add request
+            request_types_and_models(dict): a key is request type
+            and a value is a specific model
+            data(dict):
+        Raises:
+            KeyError: The request type is not defined in model
+            TypeError: The request type doesn't match with model
+        Returns:
+            None
         """
-        self.id = _id
-        self.state = state
-        self.reason = reason
-        self.state_history = state_history
-        self.from_index = from_index
-        self.from_index_resolved = from_index_resolved
-        self.bundles = bundles
-        self.removed_operators = removed_operators
-        self.organization = organization
-        self.binary_image = binary_image
-        self.binary_image_resolved = binary_image_resolved
-        self.index_image = index_image
-        self.request_type = request_type
-        self.arches = arches
-        self.bundle_mapping = bundle_mapping
-        self.omps_operator_version = omps_operator_version
+
+        if data["request_type"] not in request_types_and_models:
+            raise KeyError("Unsupported request type: %s" % data["request_type"])
+
+        if (
+            cls._accepted_request_type != data["request_type"]
+            and cls.__name__ != IIBBuildDetailsModel.__name__
```

Comment:
I don't understand why this second part is needed? IIBBuildDetailsModel is not supposed to be instantiated, so it can fail anyway

Reply:
We do not want to fail in when class is IIBBuildDetailsModel. If we fail, in this case, wont create object for any model. However IIBBuildDetailsModel is general class and should create object from one of supported classes.